src/misc/lv_mem.c

```diff
@@ -182,13 +182,14 @@ void lv_mem_free(void * data)
 void * lv_mem_realloc(void * data_p, size_t new_size)
 {
     MEM_TRACE("reallocating %p with %lu size", data_p, (unsigned long)new_size);
-    if(new_size == 0) {
+    if(data_p != NULL && new_size == 0) {
         MEM_TRACE("using zero_mem");
         lv_mem_free(data_p);
         return &zero_mem;
     }
-
-    if(data_p == &zero_mem) return lv_mem_alloc(new_size);
+    if(data_p == &zero_mem && new_size != 0) {
```

Comment:
We need to check if it is NULL (first try) or if it is zero mem.

`if((data_p == &zero_mem || data_p == NULL) && new_size != 0) {`

Reply:
Resolved in 0e0bebd